src/routes/group-routes.ts

```diff
@@ -1,16 +1,17 @@
 import express from 'express';
 
 import GroupController from '../controllers/GroupController';
+import authenticate from '../middleware/authenticate';
 import GroupService from '../services/Group.service';
 
 
 const groupRouter = express.Router();
 const groupController = new GroupController(new GroupService());
 
-groupRouter.get('/:id', groupController.getGroupById);
-groupRouter.get('/', groupController.getAllGroups);
-groupRouter.post('/', groupController.createNewGroup);
-groupRouter.put('/:id', groupController.updateGroupById);
-groupRouter.delete('/:id', groupController.removeGroupById);
+groupRouter.get('/:id', authenticate, groupController.getGroupById);
```

Comment:
It would be a little bit better if you place your auth middleware in app.ts. 

```
app.use('/users', authenticate, routes);
app.use('/groups', authenticate, groupRouter);
```